p2p/src/transport.rs

```diff
@@ -140,6 +140,32 @@ where
     ///
     /// # Errors
     ///
-    /// * If resource allocation fails for lack of priviliges or being not available.
+    /// * If resource allocation fails for lack of privileges or being not available.
     fn bind(self, bind_info: BindInfo<A>) -> Result<(Self::Endpoint, Self::Incoming)>;
 }
+
+/// Types that can be cloned where success is not guaranteed can implement this
+/// trait.
+pub trait TryClone: Sized {
```

Comment:
So concretely, do you mean creating a new crate in the tendermint-rs repo called `std-ext`? Or perhaps `tendermint-std-ext` to avoid potential crate naming clashes in future?

Reply:
Yeah that was my meaning indeed, sorry for leaving it up for interpretation.

Reply:
Cool, I can kick off a new crate :+1: Any particular naming preferences? I feel like `tendermint-std-ext` is probably a good bet.